pkgs/development/python-modules/qcodes/default.nix

```
@@ -97,6 +109,34 @@ buildPythonPackage rec {
     xarray
   ] ++ lib.optionals (pythonOlder "3.10") [ importlib-metadata ];
 
+  passthru.optional-dependencies = {
```

```suggestion
  optional-dependencies = {
```
We no longer need `passthru`.
https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/python.section.md#buildpythonpackage-parameters-buildpythonpackage-parameters